govuk_chat_evaluation/retrieval/evaluate.py

```diff
@@ -0,0 +1,220 @@
+from pathlib import Path
+from typing import Any
+
+from pydantic import BaseModel
+from sklearn.metrics import (
+    precision_score,
+    recall_score,
+    f1_score,
+    fbeta_score,
+)
+from tabulate import tabulate
+import numpy as np
+from collections.abc import Callable
+
+from ..file_system import jsonl_to_models, write_csv_results
+import logging
+
+DECIMAL_PLACES = 4
+
+
+class EvaluationResult(BaseModel):
+    question: str
+    expected_exact_paths: list[str]
+    actual_exact_paths_and_scores: list[dict[str, Any]]
+
+    @property
+    def actual_exact_paths(self) -> list[str]:
+        return [item["exact_path"] for item in self.actual_exact_paths_and_scores]
+
+    @property
+    def all_paths(self) -> list[str]:
+        return list(set(self.expected_exact_paths + self.actual_exact_paths))
+
+    @property
+    def y_true(self) -> list[int]:
+        return [int(path in self.expected_exact_paths) for path in self.all_paths]
+
+    @property
+    def y_pred(self) -> list[int]:
+        return [int(path in self.actual_exact_paths) for path in self.all_paths]
+
+    @property
+    def false_positive_cases(self) -> list[dict[str, float]]:
+        return [
+            {item["exact_path"]: item["weighted_score"]}
+            for item in self.actual_exact_paths_and_scores
+            if item["exact_path"] not in self.expected_exact_paths
+        ]
+
+    @property
+    def false_negative_cases(self) -> list[dict[str, float]]:
+        return [
+            {path: float("nan")}
+            for path in self.expected_exact_paths
+            if path not in self.actual_exact_paths
+        ]
+
+    @property
+    def true_positive_cases(self) -> list[str]:
+        return [
+            path
+            for path in self.expected_exact_paths
+            if path in self.actual_exact_paths
+        ]
```

Comment:
```suggestion
    def true_positive_cases(self) -> list[dict[str, float]]:
        return [
            {item["exact_path"]: item["weighted_score"]}
            for item in self.actual_exact_paths_and_scores
            if item["exact_path"] in self.expected_exact_paths
        ]
```


I think you forgot to return the score for the true positive cases, so a `list[dict[str, float]]` should be the expected type for true_positive_cases.